### DISCUSSION TITLE
Maintainer-contributor meet & greets

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/membership#136] [Ideas]
What do folks think of regular (maybe quarterly) maintainer meet & greets where any established project has 1-2 minutes to explain the project, explain what they are wanting to do and what kind of help they would want. Ideally the audience would be people who are interested in contributing to a project with the hopes of becoming a maintainer.

I think humanizing who a prospective contributor would be working with may make it more compelling to stick with the work. Open source feels a bit daunting at first when it's all new and it's a bunch of strangers.

There is still the hurdle with finding those prospective folks and getting them to show up. On the other hand, we could potentially cut these videos up and hand them off to the maintainers to link to within their contributing docs? I'm not 100% sure, this is a jumping off point.

**Comment by ryancheley** [comment]
@tim-schilling this sounds like a good idea. Do you think there would be an opportunity to coordinate this with the [Djangonaut.space](https://djangonaut.space/)program at all? If we were able to, I’m not sure if it would make more sense to have this meet and greet at the start or the end of the program. I’m kind of leaning towards the end of the program, but I could see it being beneficial on either side.

**Reply by tim-schilling** [reply]
I think it's possible. I'd suggest doing it a few weeks to a month after the program ends. There's always a lull that follows it because it's pretty high energy.

**Comment by nanorepublica** [comment]
I wonder if this could work in an async context? ie a maintainer records a loom video (or similar tech - clarityflow might be an interesting option here but $$), then this get's hosted in a central repo or just on the project itself as an issue?

**Reply by tim-schilling** [reply]
I think this could be an interesting addition to a suggested practices.

Ido think there's some value in having folks in the same place synchronously because of the connection it creates. It also forces a time limit on the videos too which may be useful to maintainers as a limiter to effort, making it more approachable.

**Reply by nanorepublica** [reply]
Yes, this would definitely be an addition, getting pre-recorded videos could help the synchronous conversation in terms of keeping to time? 

Having the asynchronous would allow for maintainers from disparate timezones when meeting at the time might be impossible. 

A side note, I had a similar idea for having a repository of videos that create the human connection between different OSS people.

**Comment by bckohan** [comment]
Great idea, would participate!

**Comment by tim-schilling** [comment]
I think the other concern is making sure that a repository is ready to accept new contributors. I would view this as similar to preparing your library for sprints at a DjangoCon. The 2022 DjangoCon US site has a really good post on this topic: https://2022.djangocon.us/news/dev-sprint-leaders/

Maybe we can create a checklist from that as prep work for a maintainer before allowing them to sign up for new contributors?

**Reply by nanorepublica** [reply]
I like this! But also spins up a few other ideas:

1. This might already exist or at least I think there is a discussion about it - is there a checklist for a repo to be moved into django-commons?
2. Then from this comes a larger idea that we create levels/badges for a repo (think bronze/silver/gold). Going up a level means achieving a defined goal (eg Silver could be ready to accept new contributors).
3. These levels could also be used as an endorsement of other packages outside django-commons. Thinking about it this could probably be a larger discussion... one for my blog post tomorrow!

**Reply by tim-schilling** [reply]
The incoming repo requires do cover a lot of that blog post: https://github.com/django-commons/membership/blob/main/incoming_repo_requirements.md

Though perhaps not the newcomer friendly issues. I do see this as a bit different than onboarding a project. If a connection is made due to this event, there's going to be an initial motivation/inspiration to do work. If the maintainer needs to do a lot of work to get that person up to speed, there's a lost opportunity to capitalize on that motivation.

Yes it would be very helpful to do this work at any point, but doing it for any random person versus a person you've met and has expressed interest in staying around for a while is different. That's why I'd be infavor of having a separate set of requirements to participate in this event as a maintainer versus being able to move your project here.

> These levels could also be used as an endorsement of other packages outside django-commons

100%. In case it wasn't clear, this probably shouldn't be limited to django-commons projects either.

**Reply by ryancheley** [reply]
@nanorepublica I really like this idea!

> 2. Then from this comes a larger idea that we create levels/badges for a repo (think bronze/silver/gold). Going up a level means achieving a defined goal (eg Silver could be ready to accept new contributors).

the hardest past, as always, is being able to determine how to achieve a specific goal. Also, we’d probably need to think about how / if a package slides backward (i.e. they no longer meet the criteria for Silver) and how to handle that. I don’t think it would happen, but it’s better to know how we’re going to deal with it in case it does.

**Reply by bckohan** [reply]
@nanorepublica @ryancheley 

I've been thinking about this too! Seems to me you'd want the criteria to be something you could verify in an automated way. Otherwise I can see the maintenance of it becoming too much of a burden, adding/removing criteria being too difficult and things getting stale. Criteria can be easy things to determine in an automated fashion like "supports all currently supported Django versions" or harder things like "tests against all currently supported Django versions" or even harder things that involve CI security linting or temporal things like "historical record of timely support for new Django versions". You could define levels with absolute thresholds or do a relativistic rating (i.e. Gold == top 5% of all packages, etc). If the goal was to shoot for a certain set of qualities you'd want to do thresholds, but what I like about a relativistic rating scheme is there'd be less bickering about what exactly the criteria should be. Easier to change with the times because adding/removing criteria would have less of an impact. You could measure everything you could possibly think to measure and then have the community vote and determine criteria weights from that.

Seems to me you'd want to run something like this off of djangopackages ultimately, but commons seems like a great place to pilot it.

I do think standards encouragement through badging is the probably the best way to maximize ecosystem quality. Mostly carrot, very little stick. This is also a good way to give devs something more tangible than popularity signals like stars to go off of, especially when comparing alternatives. Its hard for new work to compete with established work, even when its better.

**Reply by nanorepublica** [reply]
For completion I wrote about this yesterday: https://softwarecrafts.uk/100-words/day-246

**Reply by tim-schilling** [reply]
@nanorepublica would you like to create a PR to add that link to the django commons content area: https://github.com/django-commons/membership/blob/main/README.md

**Reply by nanorepublica** [reply]
Done! https://github.com/django-commons/membership/pull/137

**Comment by oliverandrich** [comment]
Sounds like a great idea. Count me in.

**Comment by ulgens** [comment]
+1 and I'm curious if we had any of these events so far.

**Reply by tim-schilling** [reply]
There haven't been any of these events organized yet.
